### PR TITLE
feat(build): Use broccoli for ts2dart transpilation.

### PR DESCRIPTION
**Pull request by mprobst**

### CODE REVIEW

**Comment by mprobst** [comment]
@IgorMinar FYI.

File: modules/angular2/src/facade/lang.ts [hunk 1]
```diff
@@ -1,3 +1,22 @@
+type int = number;
```
Comment (mprobst) [inline modules/angular2/src/facade/lang.ts]: I think this type definition is unused at the moment (?).

In any case, I think it belongs with the `lang.ts` file that defines the values. It was also a bit odd - `.ts` files shouldn't contain `declare` statements, only `.d.ts` should.

**Comment by mprobst** [comment]
I think I've learned something about `globals.ts` and or TS declarations today :-)

File: gulpfile.js [hunk 2]
```diff
@@ -720,7 +716,8 @@ gulp.task('test.transpiler.unittest', function() {
 // Builds all Dart packages, but does not compile them
 gulp.task('build/packages.dart', function(done) {
```
Comment (alexeagle) [inline gulpfile.js]: next step would be to replace this whole task.
Ultimately the broccoli build needs to generate the browser-served JS, so we can remove preprocessors from karma and have just one build

File: Brocfile-dart.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    {pattern: /^benchmarks\//, insertion: 'web'},
+    {pattern: /^benchmarks_external\/test\//, insertion: ''},
+    {pattern: /^benchmarks_external\//, insertion: 'web'},
+    {pattern: /^example.?\//, insertion: 'web/'},
+    {pattern: /^example.?\/test\//, insertion: ''},
+    {pattern: /^[^\/]*\/test\//, insertion: ''},
+    {pattern: /^./, insertion: 'lib'},  // catch all.
+  ];
+
+  for (var i = 0; i < replacements.length; i++) {
```
Comment (alexeagle) [inline Brocfile-dart.js]: after Igor's change, this file can be changed to TypeScript :)

**Comment by alexeagle** [comment]
Looks great

**Comment by alexeagle** [comment]
I assume you've compared the dist folder before and after this change

File: gulpfile.js (hunk 2 above)
Comment (mprobst) [inline gulpfile.js]: Yes! But one step at a time.

File: Brocfile-dart.js (hunk 3 above)
Comment (mprobst) [inline Brocfile-dart.js]: Nice.

**Comment by mprobst** [comment]
Yes, I compared the results of `gulp build/transpile.dart` before and after the change with `diff -r`. There's a minor change: this shaves `modules` off of the library name, i.e. `library angular2.foo.bar;` instead of `library modules.angular2.foo.bar`. Library names shouldn't matter in Dart anyway, but this is slightly nicer.

**Comment by mprobst** [comment]
Pushed as a3decad.

**Pull request closed without merging**